torchdynamo/guards.py

```diff
@@ -136,6 +141,23 @@ def __init__(
         self.tensor_check_names = []
         self.tensor_check_examples = []
         self.guarded_code = guarded_code
+        self.full_guarded_code_loc = (
+            f"{full_guarded_code.co_filename}:{full_guarded_code.co_firstlineno}"
+        )
+        self.check_code_to_check_type_dict = dict()
+
+    def append_code_for_check_type(self, code, check_type, name):
```

Comment:
Is it possible to convert this in a decorator?

Reply:
Should be possible - I do find decorators harder to read, but I dont have a strong opinion on style.